CHANGELOG.md

```diff
@@ -1,4 +1,5 @@
 ## Release History
+* 0.2.4 Change handlebars dependency as peerDependency to be able to install custom supported version
```

Comment:
Is this a breaking change i.e. should be `0.3.0`? (Does it force people who upgrade to install Handlebars, or does it continue working without that in some way?)

Reply:
If this forces all users to install Handlebars, what do you think about implementing custom Handlebars support another way, by letting users pass in `require('handlebars')` as an argument? (And figuring out how [this](https://github.com/mixmaxhq/rollup-plugin-handlebars-plus/blob/master/src/index.js#L10) would need to be updated.) I think it would be nice for this to work out of the box in one way or another.

It's not too hard for the user to do an extra `npm install`, but when it comes to requiring that the user install a _particular version_ of Handlebars to work with this library, then that's a bit of work that I'd like only motivated users to have to undergo.

Reply:
I do not understand how you imagine `require('handlebars')` as a parameter. In that case why do you install handlebars@4.0.5 and use another version as a parameter.
I think peer dependency does not mean it won't work out of the box. `npm install` will notify the user to install `handlebars`, because it is a peer dependency. This will be added to package.json and from that point it will be installed automatically. I do not see why would it be a problem for anybody.

The way I see this module, it supports precompile of handlebars templates with rollup bundler. So it is kind of a link between two tools. I do not see the reason why handlebars should be a dependency instead of a peer dependency.

If you check `handlebars-loader` for `webpack`, it works in a similar way.
https://github.com/pcardune/handlebars-loader/blob/master/package.json#L18
```
"peerDependencies": {
  "handlebars": ">= 1.3.0 < 5"
},
```

Reply:
> Is this a breaking change i.e. should be `0.3.0`? (Does it force people who upgrade to install Handlebars, or does it continue working without that in some way?)

You are totally right. I modified it to 0.3.0.

Reply:
Hey @btamas, I think that https://github.com/pcardune/handlebars-loader/issues/179 shows that a library that requires a peer dependency won't work out of the box and also that folks can be confused about having to install the library itself.

> I do not understand how you imagine `require('handlebars')` as a parameter

I'm talking about something like

```js
//  Within your code.
const rollup = require('rollup');
const handlebars = require('rollup-plugin-handlebars-plus');
const rootImport = require('rollup-plugin-root-import');

rollup({
  plugins: [
    handlebars({
      handlebars: {
        // This may resolve a different Handlebars than `require('handlebars')` would do within
        // the module.
        module: require('handlebars')
      }
    })
  ]
})

// Within the library's code.
function handlebars(options) {
  // This replaces the top-level `const Handlebars = require('handlebars')`
  // so we don't even load the built-in Handlebars if the developer doesn't want to use it.
  const Handlebars = options.handlebars.module || require('handlebars');
}
```

This seems to me like it would let you override Handlebars with a custom version if you wanted to, but also that the library would continue to work out of the box. And the only downside would  be that the built-in Handlebars would be an unnecessary dependency if you wanted to install your own, but that wouldn't have any runtime consequences. Thoughts?

Reply:
This solution would fit for me. I updated my PR, please check and merge it if you find everything good.

Reply:
@wearhere Any update?

Reply:
@wearhere Is there any problem with this PR? Or do you went to holiday or what?